<p>Shared coding conventions allow teams to collaborate effectively. This rule allows to check that all class names match a provided regular
expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With default provided regular expression <code>^[A-Z][a-zA-Z0-9]*$</code>:</p>
<pre>
<code data-lang="text">class my_class {...}</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="text">class MyClass {...}</code>
</pre>

