<p>Merging collapsible <code>if</code> statements increases the codeâ€™s readability.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="php">if (condition1) {
  if (condition2) {
    ...
  }
}</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="php">if (condition1 &amp;&amp; condition2) {
  ...
}</code>
</pre>

