<p>The complexity of an expression is defined by the number of <code>&amp;&amp;</code>, <code>||</code> and <code>condition ? ifTrue : ifFalse</code>
operators it contains.</p>
<p>A single expressionâ€™s complexity should not become too high to keep the code readable.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default threshold value of 3</p>
<pre>
<code data-lang="php">if ((($condition1 &amp;&amp; $condition2) || ($condition3 &amp;&amp; $condition4)) &amp;&amp; $condition5) { ... }</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="php">if ( (my_first_condition() || my_second_condition()) &amp;&amp; my_last_condition()) { ... }</code>
</pre>

