<h2>Why is this an issue?</h2>
<p>Redundant Boolean literals should be removed from expressions to improve readability.</p>
<h3>Noncompliant code example</h3>
<pre>
if ($booleanVariable == true) { /* ... */ }
if ($booleanVariable != true) { /* ... */ }
if ($booleanVariable || false) { /* ... */ }
doSomething(!false);
</pre>
<h3>Compliant solution</h3>
<pre>
if ($booleanVariable) { /* ... */ }
if (!$booleanVariable) { /* ... */ }
if ($booleanVariable) { /* ... */ }
doSomething(true);
</pre>
<h3>Exceptions</h3>
<p>The use of literal booleans in comparisons which use identity operators (<code>===</code> and <code>!==</code>) are ignored.</p>

