<h2>Why is this an issue?</h2>
<p>Return of boolean literal statements wrapped into <code>if-then-else</code> ones should be simplified.</p>
<h3>Noncompliant code example</h3>
<pre>
if (expression) {
  return true;
} else {
  return false;
}
</pre>
<h3>Compliant solution</h3>
<pre>
return expression;
</pre>

