<p>Return of boolean literal statements wrapped into <code>if-then-else</code> ones should be simplified.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="text">if (expression) {
  return true;
} else {
  return false;
}</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="text">return expression;</code>
</pre>

