<p>Using <code>count()</code> to test for emptiness works, but using <code>empty()</code> makes the code more readable and can be more performant</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="php">if (count($a) &gt; 0) { // Noncompliant
 echo $a[0];
}</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="php">if (!empty($a)) {
  echo $a[0];
}</code>
</pre>

