<h2>Why is this an issue?</h2>
<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all constant names match a provided regular expression.</p>
<h3>Noncompliant code example</h3>
<p>With the default regular expression <code>^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$</code>:</p>
<pre>
define("const1", true);

class Foo {
    const const2 = "bar";
}
</pre>
<h3>Compliant solution</h3>
<pre>
define("CONST1", true);

class Foo {
    const CONST2 = "bar";
}
</pre>

