<h2>Why is this an issue?</h2>
<p>There are several reasons for a method not to have a method body:</p>
<ul>
  <li> It is an unintentional omission and should be fixed to prevent unexpected behavior in production. </li>
  <li> It is not yet, or never will be, supported. In this case an <code>UnsupportedOperationException</code> should be thrown. </li>
  <li> The method is an intentionally-blank override. In this case a nested comment should explain the reason for the blank override. </li>
</ul>
<h3>Noncompliant code example</h3>
<pre>
public function doSomething() {
}

public function doSomethingElse() {
}
</pre>
<h3>Compliant solution</h3>
<pre>
public function doSomething() {
  // Do nothing because of X and Y.
}

public function doSomethingElse() {
  throw new UnsupportedOperationException();
}
</pre>
<h3>Exceptions</h3>
<p>Empty methods in abstract classes</p>
<pre>
abstract class Animal {
  public function speak() {}  // default implementation ignored
}
</pre>

