<h2>Why is this an issue?</h2>
<p>While not technically incorrect, the omission of curly braces can be misleading, and may lead to the introduction of errors during maintenance.</p>
<h3>Noncompliant code example</h3>
<pre>
if (condition)  // Noncompliant
  executeSomething();
</pre>
<h3>Compliant solution</h3>
<pre>
if (condition) {
  executeSomething();
}
</pre>

