<h2>Why is this an issue?</h2>
<p>For better readability, do not put more than one statement on a single line.</p>
<h3>Noncompliant code example</h3>
<pre>
if(someCondition) doSomething();
</pre>
<h3>Compliant solution</h3>
<pre>
if(someCondition) {
  doSomething();
}
</pre>
<h3>Exceptions</h3>
<p>Anonymous functions containing a single statement are ignored.</p>
<pre>
$max_comparator = function ($v) { return $v &gt; 2; };           // Compliant
$max_comparator = function ($v) { echo $v; return $v &gt; 2; };  // Noncompliant
</pre>

