<p>When only the condition expression is defined in a <code>for</code> loop, and the initialization and increment expressions are missing, a
<code>while</code> loop should be used instead to increase readability.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="text">for (;condition;) { /*...*/ }</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="text">while (condition) { /*...*/ }</code>
</pre>

