<p>
  This rule applies whenever an <code>if</code> statement is followed by one or more <code>elseif</code> statements; the final <code>elseif</code> shall be followed by an <code>else</code> statement.
</p>

<p>
  The requirement for a final <code>else</code> statement is defensive programming.
  The <code>else</code> statement should either take appropriate action or contain a suitable comment as to why no action is taken. This is consistent with the requirement to have a final default clause in a <code>switch</code> statement.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
if (condition1) {
  do_something();
} else if (condition2) {
  do_something_else();
}
</pre>

<h2>Compliant Solution</h2>
<pre>
if (condition1) {
  do_something();
} else if (condition2) {
  do_something_else();
} else {
  throw new InvalidArgumentException('message');
}
</pre>

<h2>See</h2>
<a href="https://www.securecoding.cert.org/confluence/display/seccode/MSC01-C.+Strive+for+logical+completeness">CERT, MSC01-C</a> - Strive for logical completeness <br/>
<a href="https://www.securecoding.cert.org/confluence/display/cplusplus/MSC01-CPP.+Strive+for+logical+completeness">CERT, MSC01-CPP</a> - Strive for logical completeness <br/>
<a href="https://www.securecoding.cert.org/confluence/display/jg/64.+Strive+for+logical+completeness">CERT, MSC64-Java</a> - Strive for locical completeness <br/>

