<h2>Why is this an issue?</h2>
<p>Declaring a variable only to immediately return or throw it is considered a bad practice because it adds unnecessary complexity to the code. This
practice can make the code harder to read and understand, as it introduces an extra step that doesnâ€™t add any value. Instead of declaring a variable
and then immediately returning or throwing it, it is generally better to return or throw the value directly. This makes the code cleaner, simpler, and
easier to understand.</p>
<h3>Noncompliant code example</h3>
<pre>
function computeDurationInMilliseconds() {
  $duration = ((($hours * 60) + $minutes) * 60 + $seconds ) * 1000 ;
  return $duration;
}
</pre>
<h3>Compliant solution</h3>
<pre>
function computeDurationInMilliseconds() {
  return ((($hours * 60) + $minutes) * 60 + $seconds ) * 1000;
}
</pre>

