<p>PHPâ€™s "variable variables" feature (dynamically-named variables) is temptingly powerful, but can lead to unmaintainable code.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="php">$var = 'foo';
$$var = 'bar';      //Noncompliant
$$$var = 'hello';  //Noncompliant

echo $foo; //will display 'bar'
echo $bar; //will display 'hello'</code>
</pre>

