<p>The following predefined variables are deprecated and should be replaced by the new versions:</p>
<table>
  <colgroup>
    <col style="width: 50%;">
    <col style="width: 50%;">
  </colgroup>
  <thead>
    <tr>
      <th>Replace</th>
      <th>With</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><p>$HTTP_SERVER_VARS</p></td>
      <td><p>$_SERVER</p></td>
    </tr>
    <tr>
      <td><p>$HTTP_GET_VARS</p></td>
      <td><p>$_GET</p></td>
    </tr>
    <tr>
      <td><p>$HTTP_POST_VARS</p></td>
      <td><p>$_POST</p></td>
    </tr>
    <tr>
      <td><p>$HTTP_POST_FILES</p></td>
      <td><p>$_FILES</p></td>
    </tr>
    <tr>
      <td><p>$HTTP_SESSION_VARS</p></td>
      <td><p>$_SESSION</p></td>
    </tr>
    <tr>
      <td><p>$HTTP_ENV_VARS</p></td>
      <td><p>$_ENV</p></td>
    </tr>
    <tr>
      <td><p>$HTTP_COOKIE_VARS</p></td>
      <td><p>$_COOKIE</p></td>
    </tr>
    <tr>
      <td><p>$php_errormsg</p></td>
      <td><p>error_get_last()</p></td>
    </tr>
  </tbody>
</table>
<h2>Noncompliant Code Example</h2>
<pre>
echo 'Name parameter value: ' . $HTTP_GET_VARS["name"];
</pre>
<h2>Compliant Solution</h2>
<pre>
echo 'Name parameter value: ' . $_GET["name"];
</pre>

