<h2>Why is this an issue?</h2>
<p>Using the same value on both sides of a binary operator is a code defect. In the case of logical operators, it is either a copy/paste error and,
therefore, a bug, or it is simply duplicated code and should be simplified. In the case of bitwise operators and most binary mathematical operators,
having the same value on both sides of an operator yields predictable results and should be simplified as well.</p>
<h3>Noncompliant code example</h3>
<pre>
if ( $a == $a ) { // always true
  doZ();
}
if ( $a != $a ) { // always false
  doY();
}
if ( $a == $b &amp;&amp; $a == $b ) { // if the first one is true, the second one is too
  doX();
}
if ( $a == $b || $a == $b ) { // if the first one is true, the second one is too
  doW();
}

$j = 5 / 5; //always 1
$k = 5 - 5; //always 0
</pre>
<h3>Exceptions</h3>
<p>Left-shifting 1 onto 1 is common in the construction of bit masks, and is ignored.</p>
<pre>
$i = 1 &lt;&lt; 1; // Compliant
$j = $a &lt;&lt; $a; // Noncompliant
</pre>
<h2>Resources</h2>
<ul>
  <li> {rule:php:S1656} - Implements a check on <code>=</code>. </li>
</ul>

