<p>For better readability, do not put multiple property declarations in the same statement.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="php">&lt;?php
class Foo
{
   private $bar = 1, $bar2 = 2;
}</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="php">&lt;?php
class Foo
{
   private $bar1 = 1;
   private $bar2 = 2;
}</code>
</pre>

