<h2>Why is this an issue?</h2>
<p>There is no reason to concatenate literal strings. Doing so is an exercise in reducing code readability. Instead, the strings should be
combined.</p>
<h3>Noncompliant code example</h3>
<pre>
$msg = "Hello " . "${name}" . "!";  // Noncompliant
</pre>
<h3>Compliant solution</h3>
<pre>
$msg = "Hello ${name}!";
</pre>

