<p>There is no reason to concatenate literal strings. Doing so is an exercise in reducing code readability. Instead, the strings should be
combined.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="text">$msg = "Hello " . "${name}" . "!";  // Noncompliant</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="text">$msg = "Hello ${name}!";</code>
</pre>

