<h2>Why is this an issue?</h2>
<p>Having characters before <code>&lt;?php</code> can cause "Cannot modify header information" errors and similar problems with Ajax requests.</p>
<h3>Noncompliant code example</h3>
<pre>
test&lt;?php  //Noncompliant
// ...
</pre>
<p>and</p>
<pre>
// Noncompliant; newline before opening tag
&lt;?php
// ...
</pre>
<h3>Compliant solution</h3>
<pre>
&lt;?php
// ...
</pre>

