<h2>Why is this an issue?</h2>
<p>Just as pain is your body’s way of telling you something is wrong, errors are PHP’s way of telling you there’s something you need to fix. Neither
pain, nor PHP errors should be ignored.</p>
<h3>Noncompliant code example</h3>
<pre>
@doSomethingDangerous($password);  // Noncompliant; '@' silences errors from function call
</pre>
<h3>Compliant solution</h3>
<pre>
doSomethingDangerous($password);
</pre>

