<p>Both <code>php_sapi_name()</code> and the <code>PHP_SAPI</code> constant give the same value. But calling the method is less efficient that simply
referencing the constant.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="php">if (php_sapi_name() == 'test') { ... }</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="php">if (PHP_SAPI == 'test') { ... }</code>
</pre>

