<p>Shared coding conventions allow teams to collaborate effectively. This rule flags all Perl-style comments.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="php">$myvar; # Noncompliant; this comment should have started with "//"</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="php">$myvar; // Compliant; this comment started with "//"</code>
</pre>

