<h2>Why is this an issue?</h2>
<p>Shared coding conventions allow teams to collaborate effectively. This rule flags all Perl-style comments.</p>
<h3>Noncompliant code example</h3>
<pre>
$myvar; # Noncompliant; this comment should have started with "//"
</pre>
<h3>Compliant solution</h3>
<pre>
$myvar; // Compliant; this comment started with "//"
</pre>

