<h2>Why is this an issue?</h2>
<p>Superglobal variables are predefined variables available in all scopes throughout a script. However, accessing them directly is considered bad
practice. Instead, they should be accessed through an object or framework that handles sanitation and validation.</p>
<h3>Noncompliant code example</h3>
<pre>
$name = $_POST['name'];
</pre>
<h3>Compliant solution</h3>
<pre>
$name = $this-&gt;params()-&gt;fromPost('name');
</pre>

