<h2>Why is this an issue?</h2>
<p>Both <code>php_sapi_name()</code> and the <code>PHP_SAPI</code> constant give the same value. But calling the method is less efficient that simply
referencing the constant.</p>
<h3>Noncompliant code example</h3>
<pre>
if (php_sapi_name() == 'test') { ... }
</pre>
<h3>Compliant solution</h3>
<pre>
if (PHP_SAPI == 'test') { ... }
</pre>

