<p>A value that is incremented or decremented and then not stored is at best wasted code and at worst a bug.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="php">  $i = 0;
  $i = $i++; // Noncompliant; i is still zero</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="php">  $i = 0;
  $i++;</code>
</pre>

