<p>When the call to a function doesn’t have any side effect, what is the point of making the call if the results are ignored? In such cases, either
the function call is useless and should be dropped, or the source code doesn’t behave as expected.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="php">strlen($name); // Noncompliant; "strlen" has no side effect</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="php">$length = strlen($name);</code>
</pre>

