<p>
  If you throw a general exception type, such as ErrorException, RuntimeException or Exception in a library or framework, it forces consumers to catch all exceptions, including unknown exceptions that they do not know how to handle.
</p>
<p>
  Instead, either throw a more derived type that already exists in the language (SPL exceptions), or create your own type that derives from Exception.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
throw new Exception();  // Noncompliant
</pre>

<h2>Compliant Solution</h2>
<pre>
throw new InvalidArgumentException();
// or
throw new UnexpectedValueException();
</pre>
