<h2>Why is this an issue?</h2>
<p>Just because you <em>can</em> do something, doesn’t mean you should, and that’s the case with nested ternary operations. Nesting ternary operators
results in the kind of code that may seem clear as day when you write it, but six months later will leave maintainers (or worse - future you)
scratching their heads and cursing.</p>
<p>Instead, err on the side of clarity, and use another line to express the nested operation as a separate statement.</p>
<h3>Noncompliant code example</h3>
<pre>
function get_readable_status($is_running, $has_errors) {
  return $is_running ? "Running" : ($has_errors ? "Failure" : "Succeeded");  // Noncompliant
}
</pre>
<h3>Compliant solution</h3>
<pre>
function get_readable_status($is_running, $has_errors) {
  if ($is_running) {
    return "Running";
  }
  return $has_errors ? "Failure. " : "Succeeded ";
}
</pre>
<h3>Exceptions</h3>
<p>Exclusively chained shorthand ternary operators <code>?:</code> are excluded from this rule.</p>
<pre>
$result = $option1 ?: $option2 ?: 'default';
</pre>

