<p>If a function does not return anything, it makes no sense to use its output. Specifically, passing it to another function, or assigning its
"result" to a variable is probably a bug because such functions return nothing, which is probably not what was intended.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="php">$result = closedir($dir_handle); // Noncompliant, "closedir" does not return anything.</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="php">closedir($dir_handle);</code>
</pre>

