<p>It is highly suspicious when a value is saved for a key or index and then unconditionally overwritten. Such replacements are likely errors.</p>
<h2>Noncompliant Code Example</h2>
<pre>
$someArray[1] = "someValue";
$someArray[1] = "someOtherValue"; // The intention here was probably to write to another array key.
</pre>

