<p>When verifying that code raises an exception, a good practice is to avoid having multiple method calls inside the tested code, to be explicit about
what is exactly tested.</p>
<p>When two of the methods can raise the same <strong>checked</strong> exception, not respecting this good practice is a bug, since it is not possible
to know what is really tested.</p>
<p>You should make sure that only one method can raise the expected checked exception in the tested code.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public function testSomething()
{
  try {
    g(y(1)); // Noncompliant
    $this-&gt;fail('RuntimeException is not thrown');
  } catch (RuntimeException $e) {}
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public function testSomething()
{
  $y = y(1);
  try {
    g($y);
    $this-&gt;fail('RuntimeException is not thrown by g()');
  } catch (RuntimeException $e) {}
}
</pre>

