<p>In PHPUnit, to test that an exception is thrown in a given piece of code, the <em>expectException*()</em> methods or the
<em>expectedException*</em> annotations can be used. For such a test to succeed, something in the test method has to throw an exception with the
awaited properties. Having an assertion at the end of such a test method, means that, if the test succeeds, that assertion was never evaluated because
an exception was thrown before getting to that point.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public function testA() {
    $this-&gt;expectException(\Exception::class);
    doSomething();
    $this-&gt;assertTrue($a); // Noncompliant - This assertion might never get evaluated
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public function testA() {
    $this-&gt;expectException(\Exception::class);
    doSomething();
    $this-&gt;assertTrue($a);
}
</pre>

