<p>WordPress has a database repair and optimization mode that can be activated by setting <code>WP_ALLOW_REPAIR</code> to <code>true</code> in the
configuration.</p>
<p>If activated, the repair page can be accessed by any user, authenticated or not. This makes sense because if the database is corrupted, the
authentication mechanism might not work.</p>
<p>Malicious users could trigger this potentially costly operation repeatadly slowing down the website, and making it unavailable.</p>
<h2>Ask Yourself Whether</h2>
<ul>
  <li> The database is not currently corrupted. </li>
</ul>
<p>There is a risk if you answered yes to this question.</p>
<h2>Recommended Secure Coding Practices</h2>
<p>Itâ€™s recommended to enable automatic database repair mode only in case of database corruption. This feature should be deactivated again when the
database issue is resolved.</p>
<h2>Noncompliant Code Example</h2>
<pre>
define( 'WP_ALLOW_REPAIR', true ); // Sensitive
</pre>
<h2>Compliant Solution</h2>
<pre>
// The default value is false, so the value does not have to be expilicitly set.
define( 'WP_ALLOW_REPAIR', false );
</pre>
<h2>See</h2>
<ul>
  <li> <a href="https://owasp.org/Top10/A05_2021-Security_Misconfiguration/">OWASP Top 10 2021 Category A5</a> - Security Misconfiguration </li>
  <li> <a href="https://owasp.org/Top10/A07_2021-Identification_and_Authentication_Failures/">OWASP Top 10 2021 Category A7</a> - Identification and
  Authentication Failures </li>
  <li> <a href="https://wordpress.org/support/article/editing-wp-config-php/#automatic-database-optimizing">wordpress.org</a> - Automatic Database
  Optimizing </li>
  <li> <a href="https://owasp.org/www-project-top-ten/2017/A6_2017-Security_Misconfiguration.html">OWASP Top 10 2017 Category A6</a> - Security
  Misconfiguration </li>
</ul>

