<p><code>goto</code> is an unstructured control flow statement. It makes code less readable and maintainable. Structured control flow statements such
as <code>if</code>, <code>for</code>, <code>while</code>, <code>continue</code> or <code>break</code> should be used instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="php">$i = 0;
loop:
  echo("i = $i");
  $i++;
  if ($i &lt; 10){
    goto loop;
  }</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="php">for ($i = 0; $i &lt; 10; $i++){
  echo("i = $i");
}</code>
</pre>

